### PR TITLE
P2PW-1260 - Change scale type of buy amount and change button text

### PR DESCRIPTION
**Pull request by davran1001**

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/renbtc/ui/info/RenBtcBuyBottomSheet.kt [hunk 1]
```diff
@@ -53,8 +53,14 @@ class RenBtcBuyBottomSheet : NonDraggableBottomSheetDialogFragment() {
         with(binding) {
             val feeUsd = if (priceInUsd != null) "~$$priceInUsd" else getString(R.string.common_not_available)
             topTextView.text = getString(R.string.send_account_creation_fee_format, feeUsd)
-            amountTextView.text = priceInSol.toString()
+            amountTextView.text = getString(R.string.receive_amount_in_sol, priceInSol.toString())
+            val actionText = getString(R.string.receive_pay_and_countinue, priceInSol.toString())
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/renbtc/ui/info/RenBtcBuyBottomSheet.kt:57]: можно сразу поправить countinue на continue и заinlineнить вызов прям в setActionText

**Review by gslevinkov — APPROVED** [review]

**Review by kuznetsov — APPROVED** [review]

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import org.p2p.wallet.rpc.repository.RpcRepository
+import org.p2p.wallet.utils.toPublicKey
+
+class TokenInteractor(
+    private val rpcRepository: RpcRepository,
+    private val addressInteractor: TransactionAddressInteractor,
+    private val tokenKeyProvider: TokenKeyProvider
+) {
+
+    suspend fun close(addressToClose: String): String {
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt:16]: from the current naming I can't understand what we are closing really

File: app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        val owner = tokenKeyProvider.publicKey.toPublicKey()
+        val instruction = TokenProgram.closeAccountInstruction(
+            TokenProgram.PROGRAM_ID,
+            addressToClose.toPublicKey(),
+            owner,
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt:21]: named arg is better here

File: app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            owner
+        )
+
+        val transaction = Transaction()
+        transaction.addInstruction(instruction)
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt:26]: can be rewritten with `.apply` that is used exactly for initialising objects

File: app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        val recentBlockhash = rpcRepository.getRecentBlockhash()
+        transaction.recentBlockHash = recentBlockhash.recentBlockhash
+
+        val signers = Account(tokenKeyProvider.secretKey)
+        transaction.sign(signers)
+
+        return rpcRepository.sendTransaction(transaction)
+    }
+
+    suspend fun open(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt:37]: The same poing with naming here. What are we opening?

File: app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+        val account = Account(tokenKeyProvider.secretKey)
+        val feePayer = account.publicKey
+
+        val splDestinationAddress = addressInteractor.findSplTokenAddressData(
+            destinationAddress = tokenKeyProvider.publicKey.toPublicKey(),
+            mintAddress = mintAddress
+        )
+
+        // get address
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt:49]: better to extract this logic to method `getAddress` and remove comment

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt (hunk 2 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt:16]: For example I can inject your interactor inside my class and use it this way:
```
val interactor : TokenInteractor by inject()

interactor.open(....)
interactor.close(...)
```
it looks like we are closing and opening the interactor, but not the token itself

**Review by gslevinkov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt (hunk 3 above)
Comment (davran1001) [reply app/src/main/java/org/p2p/wallet/rpc/interactor/TokenInteractor.kt:21]: it's java class which does not support named args

**Review by gslevinkov — APPROVED** [review]

**Pull request merged**